### ISSUE TITLE
configuring XTL complex number support in tests

### ISSUE THREAD

**Issue by drew-parsons** [xtensor-stack/xsimd#942]
xsimd's tests in the `test` subdir include tests with complex number support.  Support includes XTL with complex numbers.  However it's not clear how XTL complex number support is intended to be activated.

`test/CMakeLists.txt` refers to `ENABLE_XTL_COMPLEX`, and we see that if `cmake -DENABLE_XTL_COMPLEX=On ..` is active, then cmake switches the build flags from `-std=c++11` to `-std=c++14`. The tests build with `-std=c++14`` and ctest runs and passes.

However, the test files (test_batch_complex.cpp, ,test_utils.hpp), as well as the [xsimd documentation](https://xsimd.readthedocs.io/en/latest/api/xsimd_batch_complex.html#xtl-complex-support), refer to `XSIMD_ENABLE_XTL_COMPLEX`.  But  `cmake -DENABLE_XTL_COMPLEX=On ..` does not activate any `-DXSIMD_ENABLE_XTL_COMPLEX=1` build flag for compiling the tests.

The flag can be added explicitly with `cmake -DENABLE_XTL_COMPLEX=On -DCMAKE_CXX_FLAGS="-DXSIMD_ENABLE_XTL_COMPLEX=1" ..`.  In this case the tests again build, `make VERBOSE=1` confirms `XSIMD_ENABLE_XTL_COMPLEX` is set, and ctest still runs and passes.

For comparison,  `cmake -DCMAKE_CXX_FLAGS="-DXSIMD_ENABLE_XTL_COMPLEX=1" ..`.  fails.  In this case the build flags remain at  `-std=c++11` .  The documentation confirms  `-std=c++14` is needed for XTL complex number support. The [README](https://github.com/xtensor-stack/xsimd#readme) reminds us  C++14 support is needed for XTL complex number support.

In short, it is not clear how xsimd tests should be configured for testing complex number support.  Is
```
cmake -DENABLE_XTL_COMPLEX=On ..
```
supposed to be enough (does it even test XTL COMPLEX support? or does it simply build with `-std=c++14`?), or does the complex test configuration need to be
```
cmake -DENABLE_XTL_COMPLEX=On -DCMAKE_CXX_FLAGS="-DXSIMD_ENABLE_XTL_COMPLEX=1" ..
```

If the latter case, shouldn't be `-DXSIMD_ENABLE_XTL_COMPLEX=1` be set by CMakefiles.txt when `-DENABLE_XTL_COMPLEX=On ` is used, alongside the configuration of `-std=c++14` ?

**Comment by serge-sans-paille** [comment]
Thanks for reporting this one! Should be fixed in `master` branch now.
